src-test/com/zone5cloud/retrofit/core/TestOAuthAPI.java

```diff
@@ -243,4 +244,58 @@ public void testAdhocToken() {
 			assertNotNull("Token should havea  scope", token.getScope());
 		}
 	}
+
+	@Test
+	public void testZone5Server_HasClientIdAndKey(){
+		Response<User> rsp = userApi.me().blockingFirst();
+		String clientId = rsp.raw().request().header("Api-Key");
+		String clientSecret = rsp.raw().request().header("Api-Key-Secret");
+
+		System.out.println("Client Id :"+clientId + "Client Secret : "+clientSecret);
+		assertNotNull(clientId);
+		assertNotNull(clientSecret);
+	}
+
+	@Test
+	public void testNonZone5Server_HasNoClientIdAndKey(){
+		Response<User> rsp = userApi.me().blockingFirst();
+		String clientId = rsp.raw().request().header("Api-Key");
+		String clientSecret = rsp.raw().request().header("Api-Key-Secret");
+
+		System.out.println("Client Id :"+clientId + "Client Secret : "+clientSecret);
+		assertNull(clientId);
+		assertNull(clientSecret);
+	}
+
+	@Test
+	public void testZone5Server_HasClientIdAndKey_WhenTokenRefreshed(){
+		login();
+		if(isSpecialized()) {
+			Response<OAuthToken> response = authApi.adhocAccessToken("wahooride").blockingSingle();
+			OAuthToken token = response.body();
+			assertNotNull(token);
+			String clientId = response.raw().request().header("Api-Key");
+			String clientSecret = response.raw().request().header("Api-Key-Secret");
+
+			System.out.println("Client Id :" + clientId + "Client Secret : " + clientSecret);
+			assertNotNull(clientId);
+			assertNotNull(clientSecret);
+		}
+	}
+
+	@Test
+	public void testNonZone5Server_HasNoClientIdAndKey_WhenTokenRefreshed(){
+		login();
+		if(isSpecialized()) {
+			Response<OAuthToken> rsp = authApi.adhocAccessToken("wahooride").blockingSingle();
+			OAuthToken token = rsp.body();
+			assertNotNull(token);
+			String clientId = rsp.raw().request().header("Api-Key");
+			String clientSecret = rsp.raw().request().header("Api-Key-Secret");
+
+			System.out.println("Client Id :" + clientId + "Client Secret : " + clientSecret);
+			assertNull(clientId);
+			assertNull(clientSecret);
+		}
+	}
```

Comment:
how is this test different from the last? How is it testing a non zone5 server?

Reply:
The idea behind this was to test api key and client secret are only being added when it is zone5 server (I was doing this by switching env var)